rust/main/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -438,19 +440,13 @@ where
     }
 
     async fn refresh_block_and_fee_cache(&self, tx: &TypedTransaction) -> ChainResult<()> {
-        let latest_block = self
-            .provider
-            .get_block(BlockNumber::Latest)
-            .await
-            .map_err(ChainCommunicationError::from_other)?
-            .ok_or_else(|| ProviderError::CustomError("Latest block not found".into()))?;
-        let eip1559_fee =
-            estimate_eip1559_fees(self.provider.clone(), None, &latest_block, &self.domain, tx)
+        let (eip1559_fee, latest_block) =
+            estimate_eip1559_fees(self.provider.clone(), None, &self.domain, tx)
                 .await
-                .ok();
+                .map_err(ChainCommunicationError::from_other)?;
```

Comment:
Change in behaviour: before we did not throw an error when no fee were found, we returned None for fee. Now we throw an error and exist.

Reply:
We will not fail batch in this case. Implemented in https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/5974